debezium-quarkus/extensions/outbox/runtime/src/main/java/io/debezium/outbox/quarkus/internal/EventDispatcher.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.outbox.quarkus.internal;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import io.debezium.outbox.quarkus.ExportedEvent;
+
+/**
+ * An application-scope component that is responsible for observing {@link ExportedEvent} events and when
+ * detected, persists those events to the underlying database allowing Debezium to then capture and emit
+ * those change events.
+ *
+ * @author Chris Cranford
+ */
+@ApplicationScoped
+public class EventDispatcher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventDispatcher.class);
+
+    @Inject
+    EntityManager entityManager;
+
+    public EventDispatcher() {
+    }
+
+    /**
+     * An event handler for {@link ExportedEvent} events and will be called when the event fires.
+     *
+     * @param event the exported event
+     */
+    public void onExportedEvent(@Observes ExportedEvent<?, ?> event) {
+        LOGGER.debug("An exported event was found for type {}", event.getType());
+
+        // Create an OutboxEvent object based on the ExportedEvent interface
+        final OutboxEvent outboxEvent = new OutboxEvent(
+                event.getAggregateType(),
+                (String) event.getAggregateId(),
+                event.getType(),
+                (JsonNode) event.getPayload(),
```

Comment:
TODO: added the casts here for now, just to show the change to `ExportedEvent`.

Reply:
Fixed